### PR TITLE
M3-5230: Added new payment endpoints and validation

### PR DESCRIPTION
**Pull request by bnussman**
## Description

Adds new payment endpoints with respective validation.

Updates

- `POST` `https://api.linode.com/v4/account/payments`

Adds

- `GET` `https://api.linode.com/v4/account/payment-methods`
- `POST` `https://api.linode.com/v4/account/payment-methods`
  - Add validation with `PaymentMethodSchema`
- `GET` `https://api.linode.com/v4/account/client-token`

## How to test

Test payment endpoints in`@linode/api-v4` and `@linode/validation`.

### CODE REVIEW

File: packages/api-v4/src/account/payments.ts [hunk 1]
```diff
@@ -138,3 +145,51 @@ export const saveCreditCard = (data: SaveCreditCardData) => {
...
 };
+
+/**
+ * getPaymentMethods
+ *
+ * Gets a paginatated list of all the payment methods avalible
+ * on a user's account
+ *
+ */
+export const getPaymentMethods = (params?: any) => {
```
Comment (HanaXu) [inline packages/api-v4/src/account/payments.ts:156]: Not sure if there are params for this endpoint

File: packages/api-v4/src/account/payments.ts [hunk 2]
```diff
@@ -138,3 +145,51 @@ export const saveCreditCard = (data: SaveCreditCardData) => {
...
+    setMethod('GET')
+  );
+};
+
+/**
+ * addPaymentMethod
+ *
+ * Adds a new payment method to a user's account via a nonce.
+ *
+ * @param data { object }
```
Comment (HanaXu) [inline packages/api-v4/src/account/payments.ts:183]: We should add `type` here

**Review by HanaXu — APPROVED** [review]
LGTM 👍 ! Left two non-blocking comments

File: packages/api-v4/src/account/payments.ts (hunk 1 above)
Comment (bnussman) [reply packages/api-v4/src/account/payments.ts:156]: There should be since it is API paginated.

**Review by DevDW — APPROVED** [review]
LGTM

**Pull request merged**